sandbox/apparmor/apparmor.go

```diff
@@ -439,6 +441,32 @@ func PromptingSupportedByFeatures(apparmorFeatures *FeaturesSupported) (bool, st
 	if !strutil.ListContains(apparmorFeatures.ParserFeatures, "prompt") {
 		return false, "apparmor parser does not support the prompt qualifier"
 	}
+	// In the future, the complain redirect will be added to apparmor in the
+	// kernel, and it will share the same notify framework as the listener, so
+	// the presence of the notification socket will no longer be sufficient to
+	// indicate that the system supports prompting (assuming the kernel and
+	// parser features checks above pass). When this lands, an additional
+	// directory will be added at featuresSysPath/policy/notify. At this point,
+	// there will be a file at featuresSysPath/policy/notify/user which contains
+	// a list of mediation classes for which prompting is supported. Thus:
+	// - If featuresSysPath/policy/notify does not exist, and the above checks
+	//   pass, then prompting is supported.
+	// - If featuresSysPath/policy/notify exists, but does not contain a file
+	//   called user, then prompting is not supported.
+	// - If featuresSysPath/policy/notify exists and includes a file called
+	//   user, then prompting is supported.
+	// The user file should contain a list of mediation classes which are
+	// supported, and this should at the bare minimum include "file", so check
+	// for its presence among the kernel features.
+	if strutil.ListContains(apparmorFeatures.KernelFeatures, "policy:notify") {
+		if !strutil.ListContains(apparmorFeatures.KernelFeatures, "policy:notify:user:file") {
+			return false, "the kernel does not support prompting for file access"
+			// XXX: should this error message be "apparmor kernel features do not support prompting" as well?
```

Comment:
Any thoughts on this? The error message if the other apparmor kernel features lack prompting support is "apparmor kernel features do not support prompting".

Reply:
I think it is useful to distinguish each separate case to aid in debugging but perhaps each could share a common prefix to make it easy to know this relates to prompting?